mappings/net/minecraft/client/option/GameOptions.mapping

```diff
@@ -139,10 +140,65 @@ CLASS net/minecraft/class_315 net/minecraft/client/option/GameOptions
 		ARG 1 fallbackColor
 	METHOD method_19345 getTextBackgroundColor (F)I
 		ARG 1 fallbackOpacity
+	METHOD method_24230 (Lnet/minecraft/class_2487;Ljava/lang/String;)V
+		ARG 1 line
 	METHOD method_31043 setPerspective (Lnet/minecraft/class_5498;)V
 		ARG 1 perspective
 	METHOD method_31044 getPerspective ()Lnet/minecraft/class_5498;
 	METHOD method_32594 isPlayerModelPartEnabled (Lnet/minecraft/class_1664;)Z
 		ARG 1 part
 	METHOD method_32595 isTrue (Ljava/lang/String;)Z
+		ARG 0 value
 	METHOD method_32596 isFalse (Ljava/lang/String;)Z
+		ARG 0 value
+	METHOD method_33663 saveArm (Lnet/minecraft/class_1306;)Ljava/lang/String;
+		ARG 0 arm
+	METHOD method_33664 (Lnet/minecraft/class_4060;)Ljava/lang/String;
+		ARG 0 ao
+	METHOD method_33665 saveCloudRenderMode (Lnet/minecraft/class_4063;)Ljava/lang/String;
+		ARG 0 mode
+	METHOD method_33666 accept (Lnet/minecraft/class_315$class_5823;)V
+		ARG 1 visitor
+	METHOD method_33667 (Lnet/minecraft/class_315$class_5823;Lnet/minecraft/class_3419;Ljava/lang/Float;)Ljava/lang/Float;
+		ARG 1 category
+		ARG 2 currentLevel
+	METHOD method_33671 parseList (Ljava/lang/String;)Ljava/util/List;
+		ARG 0 content
+	METHOD method_33672 loadCloudRenderMode (Ljava/lang/String;)Lnet/minecraft/class_4063;
+		ARG 0 literal
+	METHOD method_33674 loadAo (Ljava/lang/String;)Lnet/minecraft/class_4060;
+		ARG 0 value
+	METHOD method_33675 loadArm (Ljava/lang/String;)Lnet/minecraft/class_1306;
+		ARG 0 arm
+	CLASS 2
+		METHOD method_33676 find (Ljava/lang/String;)Ljava/lang/String;
+			ARG 1 key
+	CLASS 3
+		METHOD method_33677 print (Ljava/lang/String;)V
+			ARG 1 key
+	CLASS class_5823 Visitor
```

Comment:
Hmm I wonder, would this allow injection of custom options.

Reply:
Yes if you can bytecode hack `accept` to add an extra call to account for another option.
Unfortunately afaik, mixin still doesn't work nicely with these package classes, right? Kaboom time as this interface indeed is package private https://maven.fabricmc.net/docs/yarn-21w05a+build.1/net/minecraft/client/option/GameOptions.class_5823.html

Reply:
You can inject to them just fine but mixin doesn't really help with extending/implementing/creating them.

Reply:
Well, mixin has trouble calling methods from this interface, though you most likely don't need new methods in this interface that handle extra data types